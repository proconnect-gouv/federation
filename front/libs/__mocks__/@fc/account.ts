import React from 'react';

export const AccountProvider = jest.fn();

export const AccountContext = React.createContext(expect.any(Object));
