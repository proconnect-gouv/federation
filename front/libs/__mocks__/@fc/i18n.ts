export const I18nService = {
  initialize: jest.fn(),
};

export const t = jest.fn((key) => key);
