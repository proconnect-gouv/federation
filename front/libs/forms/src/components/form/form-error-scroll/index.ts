/* istanbul ignore file */

// declarative file
export * from './form-error-scroll.component';
