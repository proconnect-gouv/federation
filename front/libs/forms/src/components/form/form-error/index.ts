/* istanbul ignore file */

// declarative file
export * from './form-error.component';
