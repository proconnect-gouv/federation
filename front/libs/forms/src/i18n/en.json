{
  "Form.button.copy": "Copier",
  "Form.submit": "Submit",
  "Form.reset": "Reset the form",
  "Form.message.valid": "Validation text",
  "Form.message.required": "This field is required",
  "Form.message.requestFailed": "The message could not be sent",
  "Form.message.requiredFields": "All fields with an asterisk are required.",
  "Form.select.placeholder": "Select an option",
  "Form.textarea.maxlength": " / {maxLength} characters",
  "Form.multiple.add": "Add a new value",
  "Form.multiple.remove": "Remove this value"
}
