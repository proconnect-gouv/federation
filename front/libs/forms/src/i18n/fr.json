{
  "Form.button.copy": "Copier",
  "Form.submit": "Envoyer ces informations",
  "Form.reset": "Effacer le formulaire",
  "Form.message.valid": "Texte de validation",
  "Form.message.required": "Ce champs est obligatoire",
  "Form.message.requestFailed": "Le message n’a pas pu être envoyé",
  "Form.message.requiredFields": "Tous les champs avec un astérisque sont obligatoires.",
  "Form.select.placeholder": "Sélectionner une option",
  "Form.textarea.maxlength": " / {maxLength} caractères",
  "Form.FORM_ERROR": "Une erreur s'est produite lors de la soumission du formulaire.",
  "Form.multiple.add": "Ajouter une valeur",
  "Form.multiple.remove": "Supprimer cette valeur"
}
