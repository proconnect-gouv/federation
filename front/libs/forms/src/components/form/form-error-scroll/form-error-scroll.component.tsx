import React from 'react';
import { FormSpy } from 'react-final-form';

import { useScrollToElement } from '@fc/common';

export const FormErrorScrollComponent = React.memo(() => {
  const { scrollToElement } = useScrollToElement('.fr-message--error');

  return (
    <FormSpy
      subscription={{ modifiedSinceLastSubmit: true, submitFailed: true }}
      onChange={() => {
        scrollToElement();
      }}
    />
  );
});

FormErrorScrollComponent.displayName = 'FormErrorScrollComponent';
