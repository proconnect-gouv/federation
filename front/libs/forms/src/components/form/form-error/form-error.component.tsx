import React from 'react';

import { t } from '@fc/i18n';

import { FormErrorScrollComponent } from '../form-error-scroll';

interface FormErrorComponentProps {
  error?: string;
}

export const FormErrorComponent = React.memo(({ error }: FormErrorComponentProps) => (
  <React.Fragment>
    <div className="fr-alert fr-alert--error fr-alert--sm fr-mb-6w" role="alert">
      <h3 className="fr-alert__title">{t('Form.message.requestFailed')}</h3>
      {error && <p>{error}</p>}
    </div>
    <FormErrorScrollComponent active elementClassName=".fr-alert--error" />
  </React.Fragment>
));

FormErrorComponent.displayName = 'FormErrorComponent';
