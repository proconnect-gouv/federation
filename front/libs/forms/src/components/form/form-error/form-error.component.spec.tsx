import { render } from '@testing-library/react';

import { t } from '@fc/i18n';

import { FormErrorScrollComponent } from '../form-error-scroll';
import { FormErrorComponent } from './form-error.component';

jest.mock('../form-error-scroll/form-error-scroll.component');

describe('FormErrorComponent', () => {
  it('should match the snapshot when error is defined', () => {
    // When
    const { container, getByText } = render(<FormErrorComponent error="any-error-message-mock" />);
    const textElt = getByText('any-error-message-mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(t).toHaveBeenCalledOnce();
    expect(t).toHaveBeenCalledWith('Form.message.requestFailed');
    expect(textElt).toBeInTheDocument();
    expect(FormErrorScrollComponent).toHaveBeenCalledOnce();
    expect(FormErrorScrollComponent).toHaveBeenCalledWith(
      {
        active: true,
        elementClassName: '.fr-alert--error',
      },
      {},
    );
  });

  it('should match the snapshot when error is not defined', () => {
    // When
    const { container } = render(<FormErrorComponent error="any-error-message-mock" />);

    // Then
    expect(container).toMatchSnapshot();
  });
});
