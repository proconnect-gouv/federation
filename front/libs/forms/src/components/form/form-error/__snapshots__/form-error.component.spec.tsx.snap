// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrorComponent should match the snapshot when error is defined 1`] = `
<div>
  <div
    class="fr-alert fr-alert--error fr-alert--sm  fr-mb-6w"
    role="alert"
  >
    <h3
      class="fr-alert__title"
    >
      Form.message.requestFailed
    </h3>
    <p>
      any-error-message-mock
    </p>
  </div>
</div>
`;

exports[`FormErrorComponent should match the snapshot when error is not defined 1`] = `
<div>
  <div
    class="fr-alert fr-alert--error fr-alert--sm  fr-mb-6w"
    role="alert"
  >
    <h3
      class="fr-alert__title"
    >
      Form.message.requestFailed
    </h3>
    <p>
      any-error-message-mock
    </p>
  </div>
</div>
`;
